src/main/scala/strawman/collection/View.scala

```diff
@@ -154,14 +154,14 @@ object View extends IterableFactory[View] {
     def iterator() = underlying.iterator().flatMap(f)
   }
 
-  /** A view that concatenates elements of the underlying collection with the elements
-   *  of another collection or iterator.
+  /** A view that concatenates elements of the prefix collection or iterator with the elements
+   *  of the suffix collection or iterator.
    */
-  case class Concat[A](underlying: Iterable[A], other: IterableOnce[A]) extends View[A] {
-    def iterator() = underlying.iterator() ++ other
-    override def knownSize = other match {
-      case other: Iterable[_] if underlying.knownSize >= 0 && other.knownSize >= 0 =>
-        underlying.knownSize + other.knownSize
+  case class Concat[A](prefix: IterableOnce[A], suffix: IterableOnce[A]) extends View[A] {
+    def iterator() = prefix.iterator() ++ suffix.iterator()
```

Comment:
That's interesting. In the case when a view is created from a method other than `view` of an eager collection, the view will never be iterated again but instead eagerly drained to create the result of the method (i.e. the view is method local and does not escape). Accepting `IterableOnce` for views created in such a way should be safe, while views created by calling `view` or by direct instantiation are not. Perhaps the view constructors accepting `IterableOnce` should be made package private while constructors accepting `Iterable` could be made public.